src/main/java/org/folio/bulkops/processor/HoldingsUpdateProcessor.java

```diff
@@ -27,27 +33,52 @@
 
 @Component
 @RequiredArgsConstructor
-public class HoldingsUpdateProcessor extends AbstractUpdateProcessor<HoldingsRecord> {
+public class HoldingsUpdateProcessor extends AbstractUpdateProcessor<ExtendedHoldingsRecord> {
   private static final String ERROR_MESSAGE_TEMPLATE = "No change in value for holdings record required, associated %s item(s) have been updated.";
 
   private final HoldingsClient holdingsClient;
   private final ItemClient itemClient;
   private final RuleService ruleService;
   private final ErrorService errorService;
+  private final FolioModuleMetadata folioModuleMetadata;
+  private final FolioExecutionContext folioExecutionContext;
 
   @Override
-  public void updateRecord(HoldingsRecord holdingsRecord) {
-    holdingsClient.updateHoldingsRecord(
-      holdingsRecord.withInstanceHrid(null).withItemBarcode(null).withInstanceTitle(null),
-      holdingsRecord.getId()
-    );
+  public void updateRecord(ExtendedHoldingsRecord extendedHoldingsRecord) {
+    var tenantId = extendedHoldingsRecord.getTenantId();
+    var holdingsRecord = extendedHoldingsRecord.getEntity();
+    if (StringUtils.isNotEmpty(tenantId)) {
+      try (var ignored = new FolioExecutionContextSetter(prepareContextForTenant(tenantId, folioModuleMetadata, folioExecutionContext))) {
+        holdingsClient.updateHoldingsRecord(
+          holdingsRecord.withInstanceHrid(null).withItemBarcode(null).withInstanceTitle(null),
+          holdingsRecord.getId()
+        );
+      }
+    } else {
+      holdingsClient.updateHoldingsRecord(
+        holdingsRecord.withInstanceHrid(null).withItemBarcode(null).withInstanceTitle(null),
+        holdingsRecord.getId()
+      );
+    }
   }
 
   @Override
-  public void updateAssociatedRecords(HoldingsRecord holdingsRecord, BulkOperation operation, boolean notChanged) {
-    boolean itemsUpdated = findRuleByOption(ruleService.getRules(operation.getId()), SUPPRESS_FROM_DISCOVERY)
-      .filter(bulkOperationRule -> suppressItemsIfRequired(holdingsRecord, bulkOperationRule))
-      .isPresent();
+  public void updateAssociatedRecords(ExtendedHoldingsRecord extendedHoldingsRecord, BulkOperation operation, boolean notChanged) {
+    var tenantId = extendedHoldingsRecord.getTenantId();
+    var holdingsRecord = extendedHoldingsRecord.getEntity();
+    var bulkOperationRules = ruleService.getRules(operation.getId());
+    boolean itemsUpdated;
+    if (StringUtils.isNotEmpty(tenantId)) {
+      try (var ignored = new FolioExecutionContextSetter(prepareContextForTenant(tenantId, folioModuleMetadata, folioExecutionContext))) {
+        itemsUpdated = findRuleByOption(bulkOperationRules, SUPPRESS_FROM_DISCOVERY)
```

Comment:
Could it be extracted in private method ?

Reply:
Hi @AntonAntonich , here it is important to extract rules from the current tenant context as at this part of the code context is initialized for member tenant, member tenant does not have information about bulk operation rule for central tenant.